workers/api-server/src/http/admin.rs

```diff
@@ -551,19 +551,29 @@ async fn get_fillable_amount_and_price(
 
     let (price_tx, price_rx) = oneshot::channel();
     price_reporter_job_queue
-        .send(PriceReporterJob::PeekPrice { base_token, quote_token, channel: price_tx })
+        .send(PriceReporterJob::PeekPrice {
+            base_token: base_token.clone(),
+            quote_token: quote_token.clone(),
+            channel: price_tx,
+        })
         .map_err(internal_error)?;
 
-    let price = match price_rx.await.map_err(internal_error)? {
-        PriceReporterState::Nominal(report) => report.price,
+    let price: TimestampedPrice = match price_rx.await.map_err(internal_error)? {
+        PriceReporterState::Nominal(report) => (&report).into(),
```

Comment:
why this change? An opaque `into` is less readable than field access

Reply:
so that we can call `TimestampedPrice::decimal_correct` on the result